dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchGenericDataModelMapper.cs

```diff
@@ -0,0 +1,177 @@
+    private readonly VectorStoreRecordDefinition _vectorStoreRecordDefinition;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureAISearchGenericDataModelMapper"/> class.
+    /// </summary>
+    /// <param name="vectorStoreRecordDefinition">A <see cref="VectorStoreRecordDefinition"/> that defines the schema of the data in the database.</param>
+    public AzureAISearchGenericDataModelMapper(VectorStoreRecordDefinition vectorStoreRecordDefinition)
+    {
+        Verify.NotNull(vectorStoreRecordDefinition);
+
+        this._vectorStoreRecordDefinition = vectorStoreRecordDefinition;
+    }
+
+    /// <inheritdoc />
+    public JsonObject MapFromDataToStorageModel(VectorStoreGenericDataModel<string> dataModel)
+    {
+        Verify.NotNull(dataModel);
+
+        var storageJsonObject = new JsonObject();
+
+        // Loop through all known properties and map each from the data model json to the storage json.
+        foreach (var property in this._vectorStoreRecordDefinition.Properties)
+        {
+            if (property is VectorStoreRecordKeyProperty keyProperty)
+            {
+                var storagePropertyName = keyProperty.StoragePropertyName ?? keyProperty.DataModelPropertyName;
+                storageJsonObject.Add(storagePropertyName, dataModel.Key);
+            }
+            else if (property is VectorStoreRecordDataProperty dataProperty)
+            {
+                if (dataModel.Data != null)
+                {
+                    var storagePropertyName = dataProperty.StoragePropertyName ?? dataProperty.DataModelPropertyName;
+                    var serializedJsonNode = JsonSerializer.SerializeToNode(dataModel.Data[dataProperty.DataModelPropertyName]);
+                    storageJsonObject.Add(storagePropertyName, serializedJsonNode);
+                }
+            }
+            else if (property is VectorStoreRecordVectorProperty vectorProperty)
+            {
+                if (dataModel.Vectors != null)
+                {
+                    var storagePropertyName = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+
+                    var serializedJsonNode = JsonSerializer.SerializeToNode(dataModel.Vectors[vectorProperty.DataModelPropertyName]);
+                    storageJsonObject.Add(storagePropertyName, serializedJsonNode);
+                }
+            }
+        }
+
+        return storageJsonObject;
+    }
+
+    /// <inheritdoc />
+    public VectorStoreGenericDataModel<string> MapFromStorageToDataModel(JsonObject storageModel, StorageToDataModelMapperOptions options)
+    {
+        Verify.NotNull(storageModel);
+
+        // Create variables to store the response properties.
+        var dataProperties = new Dictionary<string, object?>();
+        var vectorProperties = new Dictionary<string, object?>();
+        string key = string.Empty;
```

Comment:
Should this be null?

Reply:
Updated to null and added a check to make sure it's provided at the end.